librad/src/meta/common.rs

```diff
@@ -1,3 +1,68 @@
+use std::fmt;
+use std::str::FromStr;
+
+use url;
+use serde::{Deserialize, Serialize};
+
 pub const RAD_VERSION: u8 = 2;
 
 pub type Label = String;
+
+#[derive(Clone, Debug, PartialEq, Deserialize, Serialize)]
+pub struct Url(url::Url);
+
+impl Url {
+    pub fn parse(s: &str) -> Result<Self, urls::ParseError> {
+        Self::from_str(s)
+    }
+}
+
+pub mod urls {
```

Comment:
Why plural here? Is it to disambiguate from `url`? Could we alias the one above differently possibly?